test/common/quic/active_quic_listener_test.cc

```diff
@@ -208,7 +208,8 @@ class ActiveQuicListenerTest : public QuicMultiVersionTest {
   }
 
   void sendCHLO(quic::QuicConnectionId connection_id) {
-    client_sockets_.push_back(std::make_unique<Socket>(local_address_, nullptr, /*bind*/ false));
+    client_sockets_.push_back(std::make_unique<Network::SocketImpl>(Network::Socket::Type::Datagram,
```

Comment:
@mattklein123 With this change, the io_handle == nullptr iff bind_to_port for listensocket, both tcp and udp. 

Now UDP and TCP listener behavior is now unified including what you pointed out